### ISSUE TITLE
Support custom resolvers for @nuxt/components

### ISSUE THREAD

**Issue by antfu** [nuxt/nuxt.js#11785]

**Comment by webfansplz** [comment]
> I believe this issue can be closed since we can extend the components dirs as well as the auto-imports. cc @danielroe

Hi. I think supporting custom component resolver makes sense. Extending the components dirs option still doesn't cover the above cases.

**Comment by NexZhu** [comment]
@Atinux By supporting resolvers, framework users can just import the resolver and use it instead of copying the dirs from the resolver source code and following their changes. For example, I',m using ElementP lus UI framework with ElementPlusResolver.

**Comment by OrbisK** [comment]
I don't think this is still relevant, as modules should be able to do add components or directories.
